packages/api/src/utilities/test/batchExecute.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-disable no-console */
+
+const runBatch = ({
+  count = 0,
+  createEntry,
+  execute,
+  numberOfEntries,
+  numberOfentriesEachBatch,
+}) => {
+  if (count >= numberOfEntries) {
+    return Promise.resolve()
+  }
+
+  const numberOfBatchEntries = Math.min(
+    numberOfEntries - count,
+    numberOfentriesEachBatch
+  )
+
+  const batchIdentifier = `${count}.batch`
+  const executeIdentifier = `${batchIdentifier}.execute`
+
+  const batchData = []
+  /* eslint-disable no-plusplus */
+  for (let index = 0; index < numberOfBatchEntries; index++) {
+    batchData[index] = createEntry(count + index)
+  }
+  /* eslint-enabke no-plusplus */
+
+  console.time(executeIdentifier)
+  return execute(batchData).then(() => {
+    console.timeEnd(executeIdentifier)
+    return runBatch({
+      count: count + numberOfBatchEntries,
```

Comment:
`numberOfEntriesInThisBatch` would have been more clear to me. Now I really had to think thrice to distinguish the different numbers :-)